src/Shared/Microsoft.DotNet.Kusto/KustoClientProvider.cs

```diff
@@ -87,6 +81,106 @@ public async Task<IDataReader> ExecuteKustoQueryAsync(KustoQuery query)
             }
         }
 
+        /// <summary>
+        /// Use this method to receive large quantities of data from Kusto in a "stream". 
+        /// This method assumes the query is returning a single schema result set. 
+        /// </summary>
+        /// <param name="query"></param>
+        /// <returns></returns>
+        public IEnumerable<object[]> ExecuteStreamableKustoQuery(KustoQuery query)
+        {
+            var client = GetProvider();
+            var properties = BuildClientRequestProperties(query);
+            properties.SetOption(ClientRequestProperties.OptionResultsProgressiveEnabled, true);
+
+            string text = BuildQueryText(query);
+
+            ProgressiveDataSet pDataSet = client.ExecuteQueryV2Async(
+                DatabaseName,
+                text,
+                properties).GetAwaiter().GetResult();
```

Comment:
This method should be async and we should await this.  I'd personally probably make it `async IAsyncEnumerable<object[]>` (because IAsyncEnumerable is cool).  Though technically it could be `async Task<IEnumerable<object[]>>`, but that's a little less flexible in the future if they ever let navigating the frames be async (or if we need to background thread it because it's blocking too much)